cpp/src/lance/io/exec/scan_test.cc

```diff
@@ -102,4 +103,28 @@ TEST_CASE("Scan move to the next batch") {
     CHECK(batch.batch->num_rows() == kBatchSize);
   }
   CHECK(num_batches == kPageLength / kBatchSize * kNumBatches);
-}
\ No newline at end of file
+}
+
+TEST_CASE("Scan with multiple readers") {
```

Comment:
is it worth adding a test for nested and/or fixed_sized_list? i forgot if these all go through the same code path / offset calculation or not.

Reply:
The offsets / actual reads happen within each `FileReader`. The `Scan` node just orchestrates the reads amount of different files.